packages/twenty-server/src/database/commands/upgrade-version/0-40/0-40-phone-calling-code.command.ts

```diff
@@ -0,0 +1,344 @@
+import { InjectRepository } from '@nestjs/typeorm';
+
+import chalk from 'chalk';
+import { getCountries, getCountryCallingCode } from 'libphonenumber-js';
+import { Command } from 'nest-commander';
+import { Repository } from 'typeorm';
+import { v4 } from 'uuid';
+
+import {
+  ActiveWorkspacesCommandOptions,
+  ActiveWorkspacesCommandRunner,
+} from 'src/database/commands/active-workspaces.command';
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import {
+  FieldMetadataEntity,
+  FieldMetadataType,
+} from 'src/engine/metadata-modules/field-metadata/field-metadata.entity';
+import { ObjectMetadataEntity } from 'src/engine/metadata-modules/object-metadata/object-metadata.entity';
+import { WorkspaceMetadataVersionService } from 'src/engine/metadata-modules/workspace-metadata-version/services/workspace-metadata-version.service';
+import { generateMigrationName } from 'src/engine/metadata-modules/workspace-migration/utils/generate-migration-name.util';
+import {
+  WorkspaceMigrationColumnActionType,
+  WorkspaceMigrationTableAction,
+  WorkspaceMigrationTableActionType,
+} from 'src/engine/metadata-modules/workspace-migration/workspace-migration.entity';
+import { WorkspaceMigrationFactory } from 'src/engine/metadata-modules/workspace-migration/workspace-migration.factory';
+import { WorkspaceMigrationService } from 'src/engine/metadata-modules/workspace-migration/workspace-migration.service';
+import { TwentyORMGlobalManager } from 'src/engine/twenty-orm/twenty-orm-global.manager';
+import { computeObjectTargetTable } from 'src/engine/utils/compute-object-target-table.util';
+import { WorkspaceMigrationRunnerService } from 'src/engine/workspace-manager/workspace-migration-runner/workspace-migration-runner.service';
+import { isDefined } from 'src/utils/is-defined';
+
+const callingCodeToCountryCode = (callingCode: string): string => {
+  if (!callingCode) return '';
+  let callingCodeWithoutPlus = callingCode;
+
+  if (callingCode.startsWith('+')) {
+    callingCodeWithoutPlus = callingCode.slice(1);
+  }
+
+  return (
+    getCountries().find(
+      (countryCode) =>
+        getCountryCallingCode(countryCode) === callingCodeWithoutPlus,
+    ) || ''
+  );
+};
+
+// console.log('33', callingCodeToCountryCode('33'));
+// console.log('+33', callingCodeToCountryCode('+33'));
+// console.log('', callingCodeToCountryCode(''));
+// console.log('FR', callingCodeToCountryCode('FR'));
+
+const isCallingCode = (callingCode: string): boolean => {
+  return callingCodeToCountryCode(callingCode) !== '';
```

Comment:
I find this one a bit confusing

Reply:
this guard checks two things :
callingCode is not ''
callingCode is part of the authorized codes from the lib  libphonenumber

Note : I could use directly `callingCodeToCountryCode(callingCode) !== ''` but it was more clear of my purpose writting down this guard